editoast/src/modelsv2/prelude/list.rs

```diff
@@ -158,6 +158,18 @@ impl<M: Model + 'static> SelectionSettings<M> {
         self.paginate_counting = paginate;
         self
     }
+
+    /// Returns a tuple (page_number, page_size) based on the provided
+    /// [SelectionSettings::limit] and [SelectionSettings::offset] settings
+    ///
+    /// The page number is 1-based.
+    ///
+    /// Panics if the limit and the offset are not set.
```

Comment:
Wouldn't it be better to have default values instead of a `panic`? Something like 10 and 1 for example?

Reply:
I don't believe the models should provide default pagination values. I'm not sure if it's obvious from it's name and doc, but I wanted this function to be understood as

> If the `limit` and `offset` or the selection settings were filled according to pagination settings, what would these settings be?

If either `limit` or `offset` are unset, then the selection settings cannot have been filled from pagination settings, so calling this function is a logic error.

If you don't want to panic here, I'd argue returning an `Option<>` would make more sense in the models, and then `expect` higher in the call stack.

Reply:
`Option` seems nice (and the caller might need to `panic` anyway, but maybe having the caller decide is better).

Reply:
Done in https://github.com/OpenRailAssociation/osrd/pull/7397/commits/634c77c4b5588c7784b2a55d0f43f0ebe98e4718